app/components/interactive_dialog_controller/interactive_dialog_controller.js

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+
+export default class InteractiveDialogController extends PureComponent {
+    static propTypes = {
+        triggerId: PropTypes.string,
+        dialog: PropTypes.object,
+        navigator: PropTypes.object,
+        theme: PropTypes.object,
+    };
+
+    componentDidUpdate(prevProps) {
+        const triggerId = this.props.triggerId;
+        if (prevProps.triggerId === triggerId) {
+            return;
+        }
+
+        const dialogData = this.props.dialog;
+        if (dialogData.trigger_id !== triggerId) {
+            return;
+        }
+
+        if (!dialogData.trigger_id || !dialogData.dialog) {
+            return;
+        }
+
+        const theme = this.props.theme;
```

Comment:
why not deconstructing this like
```js
const {dialog: dialogData, triggerId, theme} = this.props;
```

Reply:
0/5 but I like just assigning them as I need them for simple cases